quickannotator/api/v1/annotation/utils.py

```diff
@@ -123,4 +126,67 @@ def compute_actor_name(project_id: int, type: constants.NamedRayActorType) -> st
         str: The generated actor name.
     """
     timestamp = time.strftime('%Y%m%d%H%M%S')
-    return f"{project_id}_{type.value}_{timestamp}"
\ No newline at end of file
+    return f"{project_id}_{type.value}_{timestamp}"
+
+
+class GeometryOperation:
+    """
+    A class to perform geometric operations on polygons.
+    """
+
+    @staticmethod
+    def union(poly1: Polygon, poly2: Polygon) -> BaseGeometry:
+        """
+        Perform a union operation on two polygons.
+
+        Args:
+            poly1 (Polygon): The first polygon.
+            poly2 (Polygon): The second polygon.
+
+        Returns:
+            BaseGeometry: The resulting geometry after the union operation.
+
+        Raises:
+            ValueError: If either of the input polygons is not valid.
+        """
+        if not poly1.is_valid:
+            raise ValueError("First polygon is not valid")
+        if not poly2.is_valid:
+            raise ValueError("Second polygon is not valid")
+        result = poly1.union(poly2)
+
+        return result
+
+    @staticmethod
+    def difference(poly1: Polygon, poly2: Polygon, multipoly_func=constants.MultiPolygonToPolygonFuncs.MAX) -> Polygon:
+        """
+        Perform a difference operation between two polygons.
+
+        Args:
+            poly1 (Polygon): The first polygon.
+            poly2 (Polygon): The second polygon to subtract from the first.
+
+        Returns:
+            Polygon: The resulting polygon after the difference operation, or None if the result is empty.
+
+        Raises:
+            ValueError: If either of the input polygons is not valid or if the resulting polygon is invalid.
+        """
+        if not poly1.is_valid:
+            raise ValueError("First polygon is not valid")
+        if not poly2.is_valid:
+            raise ValueError("Second polygon is not valid")
+        result = poly1.difference(poly2)
+
+        if result.is_empty:
+            return None
+        if not result.is_valid:
+            raise ValueError("Resulting polygon is not valid after difference operation")
+        if result.geom_type == 'MultiPolygon':
+            # If the result is a MultiPolygon, return the largest polygon by area
+            if multipoly_func == constants.MultiPolygonToPolygonFuncs.MAX:
+                result = max(result.geoms, key=lambda geom: geom.area)
+            else:
+                raise ValueError(f"Unsupported multipolygon function: {multipoly_func}")
```

Comment:
this seems to be unnecessarily limiting, why not allow the string to define the function directly, e.g., 
```python
import builtins

try:
    func = getattr(builtins, multipoly_func)
except AttributeError:
    raise ValueError(f"Unsupported multipolygon function: {multipoly_func}")

result = func(result.geoms, key=lambda g: g.area)
```

so that the user can supply any string and it would run?